src/teleburn_address.rs

```diff
@@ -0,0 +1,93 @@
+use super::*;
+use crate::index::entry::Entry;
+use bitcoin::hashes::hex::ToHex;
+use crypto::digest::Digest;
+use crypto::sha3::Sha3;
+
+#[derive(Debug, PartialEq, Serialize)]
+pub(crate) struct EthereumTeleburnAddress {
+  pub(crate) address: String,
+}
+
+impl From<InscriptionId> for EthereumTeleburnAddress {
+  fn from(inscription_id: InscriptionId) -> Self {
+    let digest = bitcoin::hashes::sha256::Hash::hash(&inscription_id.store());
+    Self {
+      address: create_address_with_checksum(&digest[0..20].to_hex()),
```

Comment:
Wouldn't it make more sense to just use md160 as the hash function here? It's the same one that hashes public keys in P2WPKH addresses, and it also produces 20 bytes.

Reply:
There’s an implementation that @casey and rob did thats already been used to teleburn some NFTs, so i stuck with this to not invalidate that burn

Reply:
Gotcha, well, this works too!